tests/model/test_model.py

```diff
@@ -628,6 +634,40 @@ def test_modernize_message_response(self, model, messages,
         assert (model.modernize_message_response(messages)
                 == expected_messages_response)
 
+    @pytest.mark.parametrize('feature_level, to_vary_in_initial_data', [
+        (None, {}),
+        (27, {}),
+        (52, {}),
+        (53, {
+            'max_stream_name_length': 30,
+            'max_topic_length': 20,
+            'max_message_length': 5000,
+        })], ids=[
+            'Zulip_2.1.x_ZFL_None_no_restrictions',
+            'Zulip_3.1.x_ZFL_27_no_restrictions',
+            'Zulip_4.0.x_ZFL_52_no_restrictions',
+            'Zulip_4.0.x_ZFL_53_with_restrictions',
+    ])
+    def test__store_content_length_restrictions(
+            self, model, initial_data, feature_level, to_vary_in_initial_data):
+        initial_data.update(to_vary_in_initial_data)
+        model.initial_data = initial_data
+        model.server_feature_level = feature_level
+
+        model._store_content_length_restrictions()
+
+        if to_vary_in_initial_data:
```

Comment:
This kind of branching suggests we could try hoisting values into the parametrizations, but this is fine.